src/credentials/FabricTable.cpp

```diff
@@ -41,145 +42,156 @@ CHIP_ERROR FabricInfo::SetFabricLabel(const CharSpan & fabricLabel)
     return CHIP_NO_ERROR;
 }
 
+namespace {
+constexpr TLV::Tag kVendorIdTag    = TLV::ContextTag(0);
+constexpr TLV::Tag kFabricLabelTag = TLV::ContextTag(1);
+} // anonymous namespace
+
 CHIP_ERROR FabricInfo::CommitToStorage(PersistentStorageDelegate * storage)
 {
-    CHIP_ERROR err = CHIP_NO_ERROR;
+    DefaultStorageKeyAllocator keyAlloc;
 
-    char key[kKeySize];
-    ReturnErrorOnFailure(GenerateKey(mFabric, key, sizeof(key)));
+    VerifyOrReturnError(CanCastTo<uint16_t>(mRootCert.size()) && CanCastTo<uint16_t>(mICACert.size()) &&
```

Comment:
Actually we should check they are all <= 400 bytes, which is spec

Reply:
Yep, will compare to `kMaxCHIPCertLength` and assert that that is not too large.

Reply:
Done.